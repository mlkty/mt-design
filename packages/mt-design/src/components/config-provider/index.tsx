import {createContext, useCallback, useContext, useMemo} from 'react';

interface ConfigProviderProps {
    prefixCls?: string;
    children?: React.ReactNode;
}

interface ConfigContextValue {
    getPrefixCls: (name: string, customPrefixCls?: string) => string;
}

const DEFAULT_PREFIXCLS = 'mt';

const getPrefixCls = (name: string, customPrefixCls?: string) => {
    if (customPrefixCls) {
        return customPrefixCls;
    }
    return `${DEFAULT_PREFIXCLS}-${name}`;
};

const ConfigContext = createContext<ConfigContextValue>({
    getPrefixCls,
});

const ConfigProvider = (props: ConfigProviderProps) => {
    const {prefixCls} = props;

    const getPrefixCls = useCallback(
        (name: string, customPrefixCls?: string) => {
            if (customPrefixCls) {
                return customPrefixCls;
            }
            const prefix = prefixCls ? prefixCls : DEFAULT_PREFIXCLS;
            return `${prefix}-${name}`;
        },
        [prefixCls]
    );

    const contextValue = useMemo<ConfigContextValue>(() => {
        return {
            getPrefixCls,
        };
    }, [getPrefixCls]);

    return (
        <ConfigContext.Provider value={contextValue}>
            {props.children}
        </ConfigContext.Provider>
    );
};

const useConfigContext = () => useContext(ConfigContext);

export {ConfigProvider, useConfigContext};
