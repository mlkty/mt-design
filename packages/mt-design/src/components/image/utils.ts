
export const isValidImage = (src: string) => {
    return new Promise(resolve => {
        const image = new Image();
        image.onload = () => resolve(true);
        image.onerror = () => resolve(false);
        image.src = src;
    });
};
