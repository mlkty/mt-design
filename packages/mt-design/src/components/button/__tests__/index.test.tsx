import { Button } from '..';
import { ConfigProvider } from '../../../contexts/config-provider';
import { render } from '../../../../tests/utils';
import mountTest from '../../../../tests/mountTest';

describe('Button', () => {
    mountTest(Button);

    it('render children correctly', () => {
        const { container } = render(<Button>Follow</Button>);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('render loading correctly', () => {
        const { container } = render(<Button loading>button</Button>);
        expect(container.firstChild).toHaveClass('mt-button');
        expect(container.firstChild).toHaveClass('mt-button--disabled');
    });

    it('render disabled correctly', () => {
        const { container } = render(<Button disabled>button</Button>);
        expect(container.firstChild).toHaveClass('mt-button');
        expect(container.firstChild).toHaveClass('mt-button--disabled');
    });

    it('render context correctly', () => {
        const { container } = render(
            <ConfigProvider prefixCls="custom">
                <Button disabled>button</Button>
            </ConfigProvider>,
        );

        expect(container.firstChild).toHaveClass('custom-button');
        expect(container.firstChild).toHaveClass('custom-button--disabled');
    });
});
