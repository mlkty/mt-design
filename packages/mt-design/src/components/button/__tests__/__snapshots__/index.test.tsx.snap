// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render children correctly 1`] = `
<button
  class="mt-button"
  role="button"
>
  Follow
</button>
`;
